WooCommerce/Classes/POS/Presentation/Item Selector/ItemList.swift

```diff
@@ -95,18 +96,19 @@ private struct ItemListRow: View {
             if #available(iOS 18.0, *) {
                 NavigationLink(value: item) {
                     ParentProductCardView(name: parentProduct.name,
-                                          imageSource: parentProduct.productImageSource,
-                                          detailText: Localization.variationsAvailable)
+                                        imageSource: parentProduct.productImageSource,
+                                        detailText: Localization.variationsAvailable)
                 }
             } else {
                 // We should drop this when we leave iOS 17.0 behind, but due to memory leaks caused by NavigationStack.
                 // we still have to use the NavigationView approach here.
                 NavigationLink(destination: {
-                    ChildItemList(parentItem: item, title: parentProduct.name)
+                    let itemsStack = node.itemType == .products ? posModel.itemsViewState.itemsStack : posModel.couponsViewState.itemsStack
+                    ChildItemList(parentItem: item, title: parentProduct.name, itemsStack: itemsStack)
```

Comment:
`ChildItemList` doesn't need `itemsStack` at all. We could remove the whole chain of passing `node` here.

Reply:
I think we can remove node, but I think it does need the stack. It needs to know whether to use `itemsViewState` or `couponsViewState`. 

I suppose we can get away without passing for coupons/items because coupons don't have a heirarchy so will never have a child list, but the next step is to add a third itemListState for search – that will have the heirarchy, and switching on item type won't work either.

I think we can probably pass the stack in and remove the node though...

Reply:
Got it, thanks for explanation.